File: packages/py/src/airlock/domain/events.py
"""Event-bus topics and the payloads that travel on them."""

from enum import StrEnum

from pydantic import BaseModel, ConfigDict

from airlock.domain.approval import ApprovalDecision
from airlock.domain.identifiers import RequestId, RunId


class EventTopic(StrEnum):
    APPROVAL_REQUESTED = "approval.requested"
    APPROVAL_DECIDED = "approval.decided"
    RUN_COMPLETED = "run.completed"
    RUN_FAILED = "run.failed"


class ApprovalDecidedEvent(BaseModel):
    model_config = ConfigDict(frozen=True)

    run_id: RunId
    request_id: RequestId
    decision: ApprovalDecision


class RunCompletedEvent(BaseModel):
    model_config = ConfigDict(frozen=True)

    run_id: RunId

File: packages/ts/src/domain/events.ts
import type { ApprovalDecision, ApprovalRequest } from "./approval";
import type { RequestId, RunId } from "./identifiers";

/** Topics on the event bus. Approval flow plus run lifecycle. */
export const EventTopic = {
  ApprovalRequested: "approval.requested",
  ApprovalDecided: "approval.decided",
  RunCompleted: "run.completed",
  RunFailed: "run.failed",
} as const;
export type EventTopic = (typeof EventTopic)[keyof typeof EventTopic];

/** Published when the loop suspends on a gated tool call. */
export type ApprovalRequestedEvent = ApprovalRequest;

/** Published by an approver; consumed by the runner to resume. */
export interface ApprovalDecidedEvent {
  readonly runId: RunId;
  readonly requestId: RequestId;
  readonly decision: ApprovalDecision;
}

export interface RunCompletedEvent {
  readonly runId: RunId;
}

export interface RunFailedEvent {
  readonly runId: RunId;
  readonly reason: string;
}
